server/proxy/pf_server.c

```diff
@@ -470,6 +470,8 @@ static DWORD WINAPI pf_server_handle_peer(LPVOID arg)
 	pf_server_channels_free(ps);
 	LOG_INFO(TAG, ps, "freeing proxy data");
 	ArrayList_Remove(server->clients, pdata);
+	if (server->config != pdata->config)
```

Comment:
I don´t see how this might happen with current code. Is this maybe part of some unrelated/follow up change?

Reply:
The way I thought this would work is a plugin could overwrite the pdata->config pointer with its own configuration (parsed using `server_config_load_ini` or `server_config_load_buffer`). 
There's a few issues that might have to be resolved like 
* having access to `server_config_load_*` from the plugins (I just  added pf_config.h and pf_config.c to plugin's CMakelists though I guess the header should have been enough if I also linked against the proxy)
* possibly having multiple plugins overwrite the pdata->config pointer (one fix would be a clone of the server->config when the connection is being established, that way every plugin can just change the settings in the config, not the config itself; this would remove the `if` and leave the `pf_server_config_free`)